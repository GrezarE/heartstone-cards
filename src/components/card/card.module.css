.card_box {
  list-style: none;
}
.card_box img {
  width: 375px;
  height: 518px;
}
.card_box:hover {
  cursor: pointer;
  transform: scale(1.1, 1.1);
  transition: 0.5s;
}/*# sourceMappingURL=card.module.css.map */