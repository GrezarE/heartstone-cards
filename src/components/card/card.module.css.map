{"version":3,"sources":["card.module.scss","card.module.css"],"names":[],"mappings":"AAKA;EACE,gBAAA;ACJF;ADKE;EACE,YAAA;EACA,aAAA;ACHJ","file":"card.module.css"}