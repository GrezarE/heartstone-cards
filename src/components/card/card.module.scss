.card_box {
  list-style: none;
  & img {
    width: 375px;
    height: 518px;
  }
  &:hover {
    cursor: pointer;
    transform: scale(1.1, 1.1);
    transition: 0.5s;
  }
}
