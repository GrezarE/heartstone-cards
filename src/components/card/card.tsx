import React from "react";
import style from "./card.module.css";

export const Card = ({ cardImage }: any) => {
  return (
    <li className={style.card_box}>
      <img src={cardImage} alt="card" className={style.card} />
    </li>
  );
};
