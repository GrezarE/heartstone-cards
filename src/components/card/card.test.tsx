import { render } from "@testing-library/react";
import { Card } from "./card";

const test = {
  link: "https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f6ddd8c7c020bf07cac7e5f712822fd3ec96af93a4f228e9b105150094a041d0.png",
  name: "Мурлокула",
};

describe("Card", () => {
  it("Карта с ссылкой", () => {
    const card = render(<Card cardImage={test.link} cardName={test.name} />);
    expect(card).toMatchSnapshot();
  });
  it("Карта без ссылки", () => {
    const card = render(<Card cardImage=" " cardName="Карта" />);
    expect(card).toMatchSnapshot();
  });
});
