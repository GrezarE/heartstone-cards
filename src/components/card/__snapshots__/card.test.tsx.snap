// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Карта 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card_box"
      >
        <img
          alt="Мурлокула"
          class="card"
          src="https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f6ddd8c7c020bf07cac7e5f712822fd3ec96af93a4f228e9b105150094a041d0.png"
        />
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card_box"
    >
      <img
        alt="Мурлокула"
        class="card"
        src="https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f6ddd8c7c020bf07cac7e5f712822fd3ec96af93a4f228e9b105150094a041d0.png"
      />
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card Карта без ссылки 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card_box"
      >
        <img
          alt="Карта"
          class="card"
          src=" "
        />
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card_box"
    >
      <img
        alt="Карта"
        class="card"
        src=" "
      />
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card Карта с ссылкой 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card_box"
      >
        <img
          alt="Мурлокула"
          class="card"
          src="https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f6ddd8c7c020bf07cac7e5f712822fd3ec96af93a4f228e9b105150094a041d0.png"
        />
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card_box"
    >
      <img
        alt="Мурлокула"
        class="card"
        src="https://d15f34w2p8l1cc.cloudfront.net/hearthstone/f6ddd8c7c020bf07cac7e5f712822fd3ec96af93a4f228e9b105150094a041d0.png"
      />
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
